bodhi/tests/server/services/test_releases.py

```diff
@@ -251,6 +254,59 @@ def test_edit_release(self):
         r = self.db.query(Release).filter(Release.name == name).one()
         self.assertEquals(r.state, ReleaseState.current)
 
+    def test_edit_mail_template(self):
+        """Test `mail_template` is saved correctly in db after release edit."""
+        name = u"F22"
+
+        res = self.app.get('/releases/%s' % name, status=200)
+        r = res.json_body
+
+        r["edited"] = name
+        r["mail_template"] = "fedora_modular_errata_template"
+        r["csrf_token"] = self.get_csrf_token()
+
+        res = self.app.post("/releases/", r, status=200)
+
+        r = self.db.query(Release).filter(Release.name == name).one()
+        self.assertEqual(r.mail_template, "fedora_modular_errata_template")
+
+    def test_edit_mail_template_with_null_value(self):
+        """Test that null value for `mail_template` gets replaced by default value."""
+        name = u"F22"
+
+        res = self.app.get('/releases/%s' % name, status=200)
+        r = res.json_body
+
+        r["edited"] = name
+        r["mail_template"] = ""
+        r["csrf_token"] = self.get_csrf_token()
+
+        res = self.app.post("/releases/", r, status=200)
+
+        r = self.db.query(Release).filter(Release.name == name).one()
+        self.assertEqual(r.mail_template, "fedora_errata_template")
+
+    def test_edit_mail_template_with_invalid_value(self):
+        """Test appropriate error is returned when provided `mail_template` doesn't exist."""
+        name = u"F22"
+        location = config.get('mail.templates_basepath')
+        directory = get_absolute_path(location)
+        template_list = [os.path.splitext(file)[0] for file in os.listdir(directory)]
+        template_vals = ", ".join(template_list)
```

Comment:
Does this test need the four lines above? It seems like they are not used.

Reply:
Yes, to keep it compatible with Python 2 and 3. Because both versions of python add elements to list in different order and hence we cannot directly compare string versions of those.

Reply:
I hadn't noticed it was in fact used on line 308 below, so nevermind!